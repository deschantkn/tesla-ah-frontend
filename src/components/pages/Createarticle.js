/* eslint-disable react/no-unused-state */
/* eslint-disable react/prop-types */
/* eslint-disable react/destructuring-assignment */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import Editor from 'prestein-dante';
import PropTypes from 'prop-types';
import { toast } from 'react-toastify';
import { Redirect } from 'react-router-dom';
import EditorState from '../widgets/Editorstate';
import {
  uploadImage,
  createArticle,
  updateArticle,
} from '../../redux/actions/article.actions';
import { getFromEditor } from '../../utils/getArticleItemData';

export class Createarticle extends Component {
  state = {
    images: [],
    redirect: false,
    slug: '',
    noImageUploaded: true,
    noArticleCreated: true,
  };

  componentWillReceiveProps(newProps) {
    const { NewUploadedImage, NewArticle } = newProps;
    const currentUploadImagePosition = Editor.getCurrentUploadImagePos();
    if (NewUploadedImage.secure_url) {
      const { images } = this.state;
      images.push({
        position: currentUploadImagePosition,
        url: NewUploadedImage.secure_url,
      });
      this.setState({ images });
    } else {
      this.setState({ noImageUploaded: true });
    }
    if (NewArticle.article) {
      const { slug } = NewArticle.article;
      toast.success('Article published successful');
      this.setState({ redirect: true, slug });
    } else {
      this.setState({ noArticleCreated: true });
    }
  }

  render() {
    const { slug } = this.state;
    if (this.state.redirect) {
      return <Redirect to={`/articles/${slug}`} />;
    }
    return (
      <div className="container create-article-container">
        <Editor
          tags={[]}
          slug=""
          images={this.state.images}
          edtrState={EditorState}
          isUpdating={false}
          uploadImage={this.props.uploadImage}
          createArticle={this.props.createArticle}
          updateArticle={this.props.updateArticle}
          getFromEditor={getFromEditor}
        />
      </div>
    );
  }
}

Editor.propTypes = {
  uploadImage: PropTypes.func.isRequired,
  createArticle: PropTypes.func.isRequired,
  edtrState: PropTypes.instanceOf(Object).isRequired,
  updateArticle: PropTypes.func.isRequired,
};

export const mapStateToProps = state => ({
  NewUploadedImage: state.article.uploadedImage,
  NewArticle: state.article.currentArticle,
});

export default connect(
  mapStateToProps,
  { updateArticle, uploadImage, createArticle },
)(Createarticle);
