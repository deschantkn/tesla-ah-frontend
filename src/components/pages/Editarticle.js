/* eslint-disable no-unused-vars */
/* eslint-disable react/prop-types */
/* eslint-disable react/destructuring-assignment */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import Editor from 'prestein-dante';
import { toast } from 'react-toastify';
import { Redirect } from 'react-router-dom';
import Preloader from '../widgets/Preloader';
import {
  getArticle,
  uploadImage,
  createArticle,
  updateArticle,
  resetProps,
} from '../../redux/actions/article.actions';
import { getFromEditor } from '../../utils/getArticleItemData';

export class Editarticle extends Component {
  state = {
    Article: {},
    slug: '',
    redirect: false,
    images: [],
  };

  componentWillMount() {
    const { slug } = this.props.match.params;
    this.setState({ slug });
    this.props.getArticle(slug);
  }

  componentWillReceiveProps(newProps) {
    const { NewUploadedImage, UpdateRes } = newProps;
    const currentUploadImagePosition = Editor.getCurrentUploadImagePos();
    if (newProps.Article) {
      this.setState({ Article: newProps.Article });
    }
    if (NewUploadedImage.secure_url) {
      toast.success('Image uploaded successfully');
      const { images } = this.state;
      images.push({
        position: currentUploadImagePosition,
        url: NewUploadedImage.secure_url,
      });
      this.setState({ images });
    }
    if (UpdateRes.message) {
      toast.success(newProps.UpdateRes.message);
      this.setState({ redirect: true });
      this.props.resetProps();
    }
  }

  render() {
    const { Article, slug } = this.state;
    if (this.state.redirect) {
      return <Redirect to="/articles" />;
    }
    if (Article.status) {
      const content = JSON.parse(Article.article.body);
      const { blocks } = content.article.body;
      const tags = Article.article.tagList;
      return (
        <div className="container create-article-container">
          <Editor
            isUpdating
            tags={tags}
            slug={slug}
            images={this.state.images}
            edtrState={{ blocks, entityMap: {} }}
            uploadImage={this.props.uploadImage}
            createArticle={this.props.createArticle}
            updateArticle={this.props.updateArticle}
            getFromEditor={getFromEditor}
          />
        </div>
      );
    }
    return (
      <div>
        <Preloader />
      </div>
    );
  }
}

const mapStateToProps = state => ({
  Article: state.article.article,
  UpdateRes: state.article.updatedArticle,
  NewUploadedImage: state.article.uploadedImage,
});
export default connect(
  mapStateToProps,
  {
    getArticle, updateArticle, uploadImage, createArticle, resetProps,
  },
)(Editarticle);
