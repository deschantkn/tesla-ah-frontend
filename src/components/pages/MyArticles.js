/* eslint-disable max-len */
/* eslint-disable no-unused-vars */
/* eslint-disable react/button-has-type */
/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/img-redundant-alt */
/* eslint-disable jsx-a11y/label-has-for */
/* eslint-disable react/destructuring-assignment */
/* eslint-disable react/prop-types */
import React, { Component } from 'react';
import moment from 'moment';
import { toast } from 'react-toastify';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import Pagination from 'react-pagination-library';
import { fontFace } from 'polished';
import {
  getMyArticles,
  deleteArticle,
} from '../../redux/actions/article.actions';
import { setLoading, setLoaded } from '../../redux/actions/ui.actions';
import Preloader from '../widgets/Preloader';
import Image from '../items/Imageitem';
import { getItemDataFromDatabase } from '../../utils/getArticleItemData';

export class MyArticles extends Component {
  state = {
    currentPage: 1,
  };

  componentDidMount() {
    this.props.getMyArticles(this.state.currentPage);
    toast.success('Fetching Articles succeeded...');
  }

  changeCurrentPage = (numPage) => {
    this.setState({ currentPage: numPage });
    this.props.getMyArticles(this.state.currentPage);
  };

  render() {
    const foundArticles = this.props.articles.data;
    // console.log('foundArticles in redux', foundArticles);
    if (foundArticles === undefined) {
      return (
        <div>
          <div className="homecontainer" data-test="homeComponent">
            <Preloader />
          </div>
        </div>
      );
    }
    if (foundArticles) {
      const articles = foundArticles.foundArticles;
      return (
        <div className="container mt-5 mb-5 center my-articles-container">
          <br />
          <div className="row ml-2">
            <div className="col-lg-12">
              <div
                htmlFor="story_title"
                className="h3 font-weight-bold text-dark"
              >
                { foundArticles.count !== 0 ? 'My Stories' : (
                  <div className="row">
                    <div className="col-lg-3" />
                    <div className="col-lg-6 text-center text-third">
                      <h2>You don&apos;t have any article so far!</h2>
                    </div>
                    <div className="col-lg-3" />
                  </div>
                )}
              </div>
            </div>
          </div>
          {articles.map((article) => {
            const {
              image,
              body,
              article: {
                title, slug, createdAt, readtime,
              },
            } = getItemDataFromDatabase(article);
            return (
              <Link
                to={`/articles/${article.slug}`}
                style={{ textDecoration: 'none' }}
              >
                <div className="row mt-5 mb-3 ml-3 " key={article.slug}>
                  <div className="col-lg-7" style={{ cursor: 'pointer' }}>
                    <div className="row">
                      <label
                        htmlFor="story_title"
                        className="h5 text-dark font-weight-bold"
                      >
                        {title}
                      </label>
                    </div>
                    <div className="row">
                      <label
                        htmlFor="story_description"
                        className="text-secondary font-weight-light"
                      >
                        {body}
                      </label>
                    </div>
                    <div className="row">
                      <div className="col-lg-0">
                        <label
                          htmlFor="story_date"
                          className="h6 text-secondary font-weight-bold"
                        >
                          {moment(createdAt).format('DD MMMM YYYY')}
                        </label>
                      </div>
                      <div className="col-lg-6">
                        <label
                          htmlFor="story_readtime"
                          className="h6 text-secondary font-weight-normal story_readtime"
                        >
                          {readtime}
                        </label>
                      </div>
                      <div className="col-lg-8" />
                    </div>
                  </div>
                  <Image title={title} image={image} />
                </div>
              </Link>
            );
          })}
          <div className="text-center container pagination-container">
            <Pagination
              currentPage={this.state.currentPage}
              totalPages={Math.ceil(foundArticles.count / 5)}
              changeCurrentPage={this.changeCurrentPage}
            />
          </div>
        </div>
      );
    }
    return (
      <div>
        <div className="homecontainer" data-test="homeComponent">
          <Preloader />
        </div>
      </div>
    );
  }
}
export const mapStateToProps = state => ({
  articles: state.article.myarticles,
  ui: state.ui,
  login: state.login,
});
export default connect(
  mapStateToProps,
  {
    getMyArticles,
    setLoaded,
    setLoading,
    deleteArticle,
  },
)(MyArticles);
