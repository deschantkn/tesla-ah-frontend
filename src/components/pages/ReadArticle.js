/* eslint-disable react/no-unused-state */
/* eslint-disable react/prop-types */
/* eslint-disable react/destructuring-assignment */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import DisplayContent from 'Dante2';
import Rater from 'react-rater';
import Moment from 'react-moment';
import { toast } from 'react-toastify';
import jwt from 'jwt-decode';
import { Link, Redirect } from 'react-router-dom';
import { getArticle, deleteArticle, resetProps } from '../../redux/actions/article.actions';
import { getUserProfile } from '../../redux/actions/author/authoruser.action';
import { DEFAULT_AVATA } from '../../utils/constants';
import Preloader from '../widgets/Preloader';


class ReadArticle extends Component {
  state = {
    Article: {},
    Author: {},
    user: {},
    slug: '',
    redirect: false,
  };

  componentWillMount() {
    const { slug } = this.props.match.params;
    let user = {};
    this.props.getArticle(slug);
    if (sessionStorage.getItem('token')) {
      user = jwt(sessionStorage.getItem('token'));
    }
    this.setState({ user, slug });
  }

  componentWillReceiveProps(newProps) {
    if (newProps.Article) {
      this.setState({ Article: newProps.Article });
      this.props.getUserProfile(newProps.Article.article.author.username);
    }
    if (newProps.Author) {
      this.setState({ Author: newProps.Author });
    }
    if (newProps.Delete.message) {
      this.setState({ redirect: true });
      toast.success('Articles has been deleted successfully...');
    }
  }

  render() {
    const {
      Article,
      Author,
      user: { username },
    } = this.state;
    let contentBlocks = [];
    if (this.state.redirect) {
      return <Redirect to="/articles" />;
    }
    if (Article && Author.profile) {
      const content = JSON.parse(Article.article.body);
      const { blocks } = content.article.body;
      contentBlocks = blocks.splice(1, blocks.length);
      const { avatar, firstName, lastName } = Author.profile;
      return (
        <div className="container view-article-content mt-5">
          <section className="editor-main-section">
            <main className="editor-main row mt-6 mb-5">
              <div className="col-lg-9 left-nav">
                <div className="title-content">
                  <p>
                    <strong>{blocks[0].text}</strong>
                  </p>
                </div>
                <div className="row profile-content ml-1">
                  <div className="">
                    <img className="" src={avatar || DEFAULT_AVATA} alt="" />
                  </div>
                  <div className="ml-3">
                    <div>
                      <strong>{`${firstName}  ${lastName}`}</strong>
                    </div>
                    <div>
                      <Moment format="D MMM YYYY">
                        {Article.article.createdAt}
                      </Moment>
                    </div>
                    <Rater total={5} rating={2} />
                  </div>
                </div>
                <div className="mt-3">
                  <DisplayContent
                    content={{ blocks: contentBlocks, entityMap: {} }}
                    read_only
                  />
                </div>
              </div>
              <div className="col-lg-1 rigth-nav text-center">
                <div className="social-buttons">
                  <div className="flauting-buttons mt-3 facebook">
                    <i className="fab fa-facebook-f" />
                  </div>
                  <div className="flauting-buttons mt-3 twitter">
                    <i className="fab fa-twitter" />
                  </div>
                  <div className="flauting-buttons mt-3 email">
                    <i className="fas fa-envelope-open" />
                  </div>
                  <div
                    className={`${
                      username === this.state.Article.article.author.username
                        ? 'hide-button'
                        : ''
                    } flauting-buttons mt-3 bookmark`}
                  >
                    <i className="far fa-bookmark" />
                  </div>
                  <div
                    className={`${
                      username === this.state.Article.article.author.username
                        ? 'hide-button'
                        : ''
                    } flauting-buttons mt-3`}
                  >
                    <i className="fas fa-thumbs-up" />
                  </div>
                  <div
                    className={`${
                      username === this.state.Article.article.author.username
                        ? 'hide-button'
                        : ''
                    } flauting-buttons mt-3 dislike`}
                  >
                    <i className="fas fa-thumbs-down" />
                  </div>
                  <Link to={`/article/edit/${this.state.slug}`}>
                    <div
                      className={`${
                        username !== this.state.Article.article.author.username
                          ? 'hide-button'
                          : ''
                      } flauting-buttons mt-3 edit`}
                    >
                      <i className="fas fa-edit" />
                    </div>
                  </Link>
                  <div
                    className={`${
                      username !== this.state.Article.article.author.username
                        ? 'hide-button'
                        : ''
                    } flauting-buttons mt-3 delete`}
                    data-toggle="modal"
                    data-target="#myModal"
                  >
                    <i className="fas fa-trash-alt" />
                  </div>
                </div>
              </div>
            </main>
            <div id="myModal" className="modal fade" role="dialog">
              <div className="modal-dialog">

                <div className="modal-content">
                  <div className="modal-header model-header-div">
                    <button type="button" className="close text-white" data-dismiss="modal">&times;</button>
                  </div>
                  <div className="modal-body modal-body-div">
                    <p> Are you sure you want to delete this article ?</p>
                  </div>
                  <div className="modal-footer modal-footer-div">
                    <button type="button" className="btn btn-default text-white" data-dismiss="modal">No</button>
                    <button type="button" className="btn btn-default text-white" data-dismiss="modal" onClick={() => this.props.deleteArticle(this.state.slug)}>Yes</button>
                  </div>
                </div>

              </div>
            </div>
          </section>
        </div>
      );
    }
    return (
      <div>
        <Preloader />
      </div>
    );
  }
}

const mapStateToProps = state => ({
  Article: state.article.article,
  Author: state.author.authorprofile,
  Delete: state.article.deletedArticle,
});

export default connect(
  mapStateToProps,
  {
    getArticle, getUserProfile, deleteArticle, resetProps,
  },
)(ReadArticle);
