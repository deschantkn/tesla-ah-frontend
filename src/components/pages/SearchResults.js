/* eslint-disable no-lone-blocks */
/* eslint-disable react/prefer-stateless-function */
import React, { Fragment, Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Article from '../items/Article';
import Search from '../items/Search';
import Alert from '../common/Alert';
import Spinner from '../widgets/Spinner';

export class Results extends Component {
  static propTypes = {
    search: PropTypes.instanceOf(Object).isRequired,
  }

  constructor() {
    super();
    this.state = {
      articles: [],
    };
  }

  componentWillReceiveProps(nextProps) {
    this.setState({
      articles: nextProps.search.results,
    });
  }

  render() {
    const { articles } = this.state;
    const {
      search: {
        failed, message, pending, done,
      },
    } = this.props;
    return (
      <Fragment>
        <div className="search__results">
          <Search />
          {pending ? (
            <div className="row">
              <div className="col-sm-4" />
              <div className="col-sm-4">
                <Spinner caption="Searching..." />
              </div>
              <div className="col-sm-4" />
            </div>
          ) : null}
          {!articles.length && done ? (
            <div className="container">
              <Alert type="warning">
                <p>No articles for this author...</p>
              </Alert>
            </div>
          ) : null}
          {articles.map((article) => {
            { /* const goodArticle = getFreshArticle(article);
            console.log(goodArticle); */ }
            return (
              <Article key={article.slug} article={article} />
            );
          })}
          {failed ? (
            <div className="container">
              <Alert type="warning">
                <p>{message}</p>
              </Alert>
            </div>
          ) : null}
        </div>
      </Fragment>
    );
  }
}

export const mapStateToProps = state => ({
  search: state.search,
});

export default connect(mapStateToProps)(Results);
