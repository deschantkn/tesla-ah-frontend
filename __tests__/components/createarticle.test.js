import React from 'react';
import { shallow } from 'enzyme';
import { Createarticle, mapStateToProps } from '../../src/components/pages/Createarticle';

let createArticle;
describe('Home Components tests...', () => {
  const props = {
    uploadImage: jest.fn(),
    createArticle: jest.fn(),
    updateArticle: jest.fn(),
  };
  beforeAll(() => {
    createArticle = shallow(<Createarticle {...props} />);
  });

  it('Should render Home component', () => {
    createArticle
      .instance()
      .componentWillReceiveProps({
        NewUploadedImage: { secure_url: undefined },
        NewArticle: { article: undefined },
      });
    createArticle
      .instance()
      .componentWillReceiveProps({
        NewUploadedImage: { secure_url: 'dcdcdcdcdc' },
        NewArticle: { article: { } },
      });
    expect(typeof mapStateToProps({ article: { uploadImage: { } } })).toBe('object');
    expect(createArticle).toMatchSnapshot();
  });
});
