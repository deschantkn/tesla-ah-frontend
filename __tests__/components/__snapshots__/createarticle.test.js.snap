// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Components tests... Should render Home component 1`] = `ShallowWrapper {}`;
